### PR TITLE
move pkg/util/version to staging

### PR DESCRIPTION
**Pull request by BenTheElder**
**What this PR does / why we need it**: Moves `pkg/util/version` into staging. This will let downstream tooling parse Kubernetes version strings without either reimplementing the parsing (though they do just seem to be `"v"+"<a semver>"` ...) or importing kubernetes.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**: I'll understand if we don't want to do this, but it seems extremely self-contained already and I'd love to leverage it for some test tooling to parse Kubernetes build versions (particularly pre-release).

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by BenTheElder** [comment]
/kind cleanup
I guess? this doesn't really seem to fit any of [the current kind/ labels](https://github.com/kubernetes/kubernetes/labels?utf8=%E2%9C%93&q=kind%2F) ...

**Comment by neolit123** [comment]
:+1: 

/test pull-kubernetes-e2e-gce-device-plugin-gpu

**Comment by dchen1107** [comment]
/lgtm

Thanks for picking up this work.

**Review by krzyzacy — APPROVED** [review]
/lgtm

File: test/test_owners.csv [hunk 1]
```diff
@@ -809,7 +809,7 @@ k8s.io/kubernetes/pkg/util/tail,zmerlynn,1,
 k8s.io/kubernetes/pkg/util/taints,rrati,0,
 k8s.io/kubernetes/pkg/util/term,davidopp,1,
 k8s.io/kubernetes/pkg/util/threading,roberthbailey,1,
-k8s.io/kubernetes/pkg/util/version,danwinship,0,
```
Comment (ixdy) [inline test/test_owners.csv]: I think this file is supposed to be sorted? :man_shrugging:

File: test/test_owners.csv [hunk 2]
```diff
@@ -809,7 +809,7 @@ k8s.io/kubernetes/pkg/util/tail,zmerlynn,1,
...
+k8s.io/apimachinery/pkg/util/version,danwinship,0,
```
Comment (ixdy) [inline test/test_owners.csv]: I think this file is supposed to be sorted? :man_shrugging:

Comment (BenTheElder) [reply test/test_owners.csv]: discussed offline, we don't think this file is actually maintained, at the very least this shouldn't break anything.

**Pull request merged**